deps/fdasrsf/dp_grid.c

```diff
@@ -0,0 +1,257 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "dp_nbhd.h"
+#include "dp_grid.h"
+
+
+void dp_all_edge_weights(
+    double* Q1, double* T1, int nsamps1,
+    double* Q2, double* T2, int nsamps2,
+    int dim,
+    double* tv1, int* idxv1, int ntv1,
+    double* tv2, int* idxv2, int ntv2,
+    double* W, double lam)
+{
+    int sr, sc;  /* source row and column */
+    int tr, tc;  /* target row and column */
+    int l1, l2, l3;  /* for multidimensional array mapping */
+    int i;
+
+    for (i=0; i<ntv1*ntv2*ntv1*ntv2; W[i++]=1e6);
+
+    /* W is a ntv2 x ntv1 x ntv2 x ntv1 array.
+     * Weight of edge from (tv1[i],tv2[j]) to (tv1[k],tv2[l])
+     * (Cartesian coordinates) is in grid(j,i,l,k).
+     * Mapping:
+     *  (j,i,l,k) :--> j*ntv1*ntv2*ntv1 +
+     *                 i*ntv2*ntv1 +
+     *                 l*ntv1 +
+     *                 k
+     */
+    l1 = ntv1 * ntv2 * ntv1;
+    l2 = ntv2 * ntv1;
+    l3 = ntv1;
+
+    for (tr=1; tr<ntv2; ++tr) {
```

Comment:
Spaces also around assign and comparison operators. Using `++` prefix notation indicates that this guy comes from a `C++` background, which may explain why the casts were used in `malloc` and `calloc`.